server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -325,6 +342,17 @@ public void removeField(String path) {
         removeField(path, false);
     }
 
+    @Nullable
+    private WriteField getWriteField(String path) {
+        if (path == null) {
+            return null;
+        }
+        if ((path.startsWith("$('") && path.endsWith("')")) || (path.startsWith("$(\"") && path.endsWith("\")"))) {
+            return new WriteField(path.substring(3, path.length() - 2), this::getCtxMap);
```

Comment:
Do we need some additional validation that no chars like `[` are inside or is this already handled by WriteField? If a user would use `$("foo.bar['hello']")` I assume it would just return not found or an exception? 

BTW this can also be solved through docs.